PBblas/internal/MakeR8Set.ecl

```diff
@@ -0,0 +1,56 @@
+/*##############################################################################
+## HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems�.  All rights reserved.
+############################################################################## */
+
+IMPORT $.^ as PBblas;
+IMPORT PBblas.Types;
+dimension_t := Types.dimension_t;
+value_t := Types.value_t;
+Layout_Cell := Types.Layout_Cell;
+
+/**
+  * Take a dataset of cells (or sub-matrix thereof) and pack into a dense matrix.
+  * in column first order [r1c1, r1c2, ... r1cM, r2c1, r2c2, ... rNcM].
+  * First row and first column are one based.
+  *
+  * @param  r         The number of rows in the matrix or sub-matrix
+  * @param  s         The number of columns in the matrix or sub-matrix
+  * @param first_row  The first row to include in the submatrix
+  * @param first_col  The first column to include in the submatrix
+  * @param D          The matrix to convert in Layout_Cell form
+  * @param transpose  Boolean to transpose the matrix during conversion (optional)
+  * @return           Dense matrix or sub-matrix (SET OF REAL8)
+  */
+EXPORT SET OF REAL8 makeR8Set(dimension_t r, dimension_t s,
+                              dimension_t first_row, dimension_t first_col,
+                              DATASET(Layout_Cell) D,
+                              BOOLEAN transpose=FALSE) := BEGINC++
+    // copy of Layout_Cell translated to C
```

Comment:
It would be better (not critical though) to wrap this in a #ifndef/#endif incase you needed to create another BEGINC++ attribute that took a dataset of Cell records.